control-plane/agents/src/bin/ha/cluster/switchover.rs

```diff
@@ -257,14 +257,28 @@ impl SwitchOverEngine {
             info!("Starting worker-{} of Switchover Engine", i.to_string());
             let cloned_self = self.clone();
             let cloned_channel = recv.clone();
-            tokio::spawn(async move { cloned_self.process_request(cloned_channel).await });
+            tokio::spawn(async move {
+                cloned_self
+                    .process_request(cloned_channel, i.to_string())
+                    .await
+            });
         }
     }
 
     /// Switchover request to be handled synchronously in each worker task.
-    async fn process_request(self, recv: Arc<Mutex<UnboundedReceiver<SwitchOverRequest>>>) {
+    async fn process_request(
+        self,
+        recv: Arc<Mutex<UnboundedReceiver<SwitchOverRequest>>>,
+        worker_num: String,
+    ) {
         loop {
-            if let Some(mut q) = recv.lock().await.recv().await {
+            let mut lo = recv.lock().await;
```

Comment:
It's preferred to make use of scope, example:
```
let work = {
    recv.lock().await.recv().await
};
if let Some(mut work) = work {
}
```

Reply:
I use drop as soon as the request is received. This would allow next worker in queue to acquire the lock. Can i keep this as is?

Reply:
Using scopes IMHO is more elegant, but it's up to you.

Reply:
Thinking of keeping it as is Tiago. Thanks you for the suggestion.